addons/purchase/models/purchase_bill_line_match.py

```diff
@@ -109,40 +116,55 @@ def action_open_line(self):
             'res_id': self.account_move_id.id if self.account_move_id else self.purchase_order_id.id,
         }
 
+    @api.model
+    def _action_create_bill_from_po_lines(self, partner, po_lines):
+        """ Create a new vendor bill with the selected PO lines """
+        bill = self.env['account.move'].create({
+            'move_type': 'in_invoice',
+            'partner_id': partner.id,
+        })
+
+        for po_line in po_lines.filtered(lambda pol: not pol.display_type):
+            new_line_values = po_line._prepare_account_move_line(bill)
+            bill.invoice_line_ids += self.env['account.move.line'].new(new_line_values)
+
+        return {
+            'type': 'ir.actions.act_window',
+            'res_model': 'account.move',
+            'view_mode': 'form',
+            'res_id': bill.id,
+        }
+
     def action_match_lines(self):
-        if not self.pol_id or not self.aml_id:
-            raise UserError(_("You must select at least one Purchase Order line and one Vendor Bill line to match them."))
+        if not self.pol_id:  # we need POL(s) to either match or create bill
+            raise UserError(_("You must select at least one Purchase Order line to match or create bill."))
+        if not self.aml_id:  # select POL(s) without AML -> create a draft bill with the POL(s)
+            return self._action_create_bill_from_po_lines(self.partner_id, self.pol_id)
+        if len(self.aml_id.move_id) > 1:  # for purchase matching, disallow matching multiple bills at the same time
+            raise UserError(_("You can't select lines from multiple Vendor Bill to do the matching."))
 
-        matches_found = 0
-        problem_products = self.env['product.product']
         pol_by_product = self.pol_id.grouped('product_id')
         aml_by_product = self.aml_id.grouped('product_id')
+        residual_pol_ids = self.pol_id
+        residual_aml_ids = self.aml_id
```

Comment:
```suggestion
        residual_purcharse_order_lines = self.pol_id
        residual_account_move_lines = self.aml_id
```
[Guidelines](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#symbols-and-conventions:~:text=suffix%20your%20variable,of%20res.partner)